idea/idea-frontend-fir/tests/org/jetbrains/kotlin/idea/fir/frontend/api/symbols/AbstractSymbolByPsiTest.kt

```diff
@@ -9,13 +9,29 @@ import org.jetbrains.kotlin.idea.frontend.api.KtAnalysisSession
 import org.jetbrains.kotlin.idea.frontend.api.symbols.KtSymbol
 import org.jetbrains.kotlin.psi.KtDeclaration
 import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtParameter
 import org.jetbrains.kotlin.psi.psiUtil.collectDescendantsOfType
 import org.jetbrains.kotlin.test.services.TestServices
+import java.lang.IllegalStateException
 
 abstract class AbstractSymbolByPsiTest : AbstractSymbolTest() {
     override fun KtAnalysisSession.collectSymbols(ktFile: KtFile, testServices: TestServices): List<KtSymbol> {
-        return ktFile.collectDescendantsOfType<KtDeclaration>().map { declaration ->
-            declaration.getSymbol()
+        return ktFile.collectDescendantsOfType<KtDeclaration>().mapNotNull { declaration ->
+            try {
```

Comment:
Maybe the better solution will be to check if it valid symbol or not and only then convert it to KtSymbol?

Reply:
Also, `collectDescendantsOfType` accepts predicate lambda, so we can get rid of `mapOfNull`

Reply:
Using the predicate for `collectDescendantsOfType` makes things a lot cleaner!